core/src/main/java/io/questdb/griffin/engine/join/HashJoinRecordCursorFactory.java

```diff
@@ -109,7 +108,8 @@ public boolean supportsUpdateRowId(CharSequence tableName) {
     }
 
     private void buildMapOfSlaveRecords(RecordCursor slaveCursor, SqlExecutionCircuitBreaker circuitBreaker) {
-        HashOuterJoinRecordCursorFactory.buildMap(slaveCursor, slaveCursor.getRecord(), joinKeyMap, slaveKeySink, slaveChain, circuitBreaker);
+        cursor.joinKeyMap.inflate();
+        HashOuterJoinRecordCursorFactory.buildMap(slaveCursor, slaveCursor.getRecord(), cursor.joinKeyMap, slaveKeySink, slaveChain, circuitBreaker);
```

Comment:
If exception is thrown here cursor.joinKeyMap will not be closed because `HashJoinRecordCursor.of` method not yet called.